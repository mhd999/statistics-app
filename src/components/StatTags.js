// @flow

import React from 'react';
import styler from 'react-styling';

type Props = {
    tags: string[],
};

const style = styler `
  TagStyle
    font-size: 14px
    background-color: #f2f2f2
    color: #a53d7c
    outline: none
    padding-left: 1em
    padding-right: 1em
    padding-top: 0.2em
    padding-bottom: 0.2em
    margin-top: 0.2em
    min-height: 15px
    display: inline-block
    cursor: pointer
    margin: 1px
 `;

const StatTags = (props: Props) => {

  let tagsList = props.tags.map((tag, index) => {
    return <span key={index} style={style.TagStyle}>{tag}</span>
  })
  
  return <div>{tagsList}</div>;
}

export default StatTags;