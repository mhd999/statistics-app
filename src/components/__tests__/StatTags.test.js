import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import StatTags from '../StatTags';

const tags = ['tag1', 'tag2', 'tag3'];

const props = {
  tags: tags,
};


describe('<StatTags />', () => {
  it('renders a array of tags', () => {
    const component = shallow(
      <StatTags {...props} />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
});
