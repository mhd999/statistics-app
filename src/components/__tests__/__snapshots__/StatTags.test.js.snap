exports[`<StatTags /> renders a array of tags 1`] = `
<div>
  <span
    style={
      Object {
        "backgroundColor": "#f2f2f2",
        "color": "#a53d7c",
        "cursor": "pointer",
        "display": "inline-block",
        "fontSize": "14px",
        "margin": "1px",
        "marginTop": "0.2em",
        "minHeight": "15px",
        "outline": "none",
        "paddingBottom": "0.2em",
        "paddingLeft": "1em",
        "paddingRight": "1em",
        "paddingTop": "0.2em",
      }
    }>
    tag1
  </span>
  <span
    style={
      Object {
        "backgroundColor": "#f2f2f2",
        "color": "#a53d7c",
        "cursor": "pointer",
        "display": "inline-block",
        "fontSize": "14px",
        "margin": "1px",
        "marginTop": "0.2em",
        "minHeight": "15px",
        "outline": "none",
        "paddingBottom": "0.2em",
        "paddingLeft": "1em",
        "paddingRight": "1em",
        "paddingTop": "0.2em",
      }
    }>
    tag2
  </span>
  <span
    style={
      Object {
        "backgroundColor": "#f2f2f2",
        "color": "#a53d7c",
        "cursor": "pointer",
        "display": "inline-block",
        "fontSize": "14px",
        "margin": "1px",
        "marginTop": "0.2em",
        "minHeight": "15px",
        "outline": "none",
        "paddingBottom": "0.2em",
        "paddingLeft": "1em",
        "paddingRight": "1em",
        "paddingTop": "0.2em",
      }
    }>
    tag3
  </span>
</div>
`;
